llvm/lib/Transforms/Utils/Local.cpp

```diff
@@ -3586,10 +3586,13 @@ DIExpression *llvm::getExpressionForConstant(DIBuilder &DIB, const Constant &C,
     return createIntegerExpression(C);
 
   auto *FP = dyn_cast<ConstantFP>(&C);
-  if (FP && (Ty.isFloatTy() || Ty.isDoubleTy())) {
+  if (FP &&
+      (Ty.isBFloatTy() || Ty.isHalfTy() || Ty.isFloatTy() || Ty.isDoubleTy())) {
     const APFloat &APF = FP->getValueAPF();
-    return DIB.createConstantValueExpression(
-        APF.bitcastToAPInt().getZExtValue());
+    APInt const &API = APF.bitcastToAPInt();
+    if (auto Temp = API.tryZExtValue())
```

Comment:
Now that we aren't supporting values >64bits, I think this can revert to using `getZExtValue`, since `tryZExtValue` only returns `std::nullopt` if `getActiveBits() <= 64`.